src/objects/api/serializers.py

```diff
@@ -1,47 +1,78 @@
-import jsonschema
-import requests
+from django.db import transaction
+
 from rest_framework import serializers
 
-from .models import Object
+from objects.core.models import Object, ObjectRecord
+
+from .validators import CorrectionValidator, JsonSchemaValidator
+
+
+class ObjectRecordSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ObjectRecord
+        fields = (
+            "data",
+            "startDate",
+            "endDate",
+            "registrationDate",
+            "correct",
+        )
+        extra_kwargs = {
+            "startDate": {"source": "start_date"},
+            "endDate": {"source": "end_date", "read_only": True},
+            "registrationDate": {"source": "registration_date", "read_only": True},
+            "correct": {"required": False},
+        }
+
+
+class HistoryRecordSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ObjectRecord
+        fields = (
+            "id",
+            "data",
+            "startDate",
+            "endDate",
+            "registrationDate",
+            "corrected",
+        )
+        extra_kwargs = {
+            "id": {"read_only": True},
+            "startDate": {"source": "start_date"},
+            "endDate": {"source": "end_date", "read_only": True},
+            "registrationDate": {"source": "registration_date", "read_only": True},
+            "corrected": {"read_only": True},
+        }
 
 
 class ObjectSerializer(serializers.HyperlinkedModelSerializer):
+    record = ObjectRecordSerializer(source="current_record")
+
     class Meta:
         model = Object
-        fields = ("url", "type", "version", "data")
+        fields = ("url", "type", "typeVersion", "record")
```

Comment:
This comment will touch several places, but in essence:

* `type` should be readonly, it doesn't make sense to change the type.
* `typeVersion` should be part of the record. People need to know the version for each record.
* can we also add the `uuid` field?

Reply:
`type` is immutable now, i.e. it can be created, but it can't be updated with a new value
`typeVersion` is moved to the record 
`uuid` field is added to Record resource and now in the API records are referenced with their uuid